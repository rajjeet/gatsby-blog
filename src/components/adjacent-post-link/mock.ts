export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
