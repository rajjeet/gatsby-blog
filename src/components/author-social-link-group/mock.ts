import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

