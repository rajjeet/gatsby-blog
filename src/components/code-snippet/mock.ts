import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
