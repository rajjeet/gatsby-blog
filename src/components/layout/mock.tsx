import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

