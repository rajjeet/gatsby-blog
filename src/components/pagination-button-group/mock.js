export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
