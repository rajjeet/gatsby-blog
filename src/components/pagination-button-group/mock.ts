export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
