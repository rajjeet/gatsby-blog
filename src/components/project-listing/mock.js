import { createMockProjects } from '../../utils/testing';

export const makeProps = () => ({
  className: {}, projects: createMockProjects, heading: '',
});
