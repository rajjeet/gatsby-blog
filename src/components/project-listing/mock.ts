import { createMockProjects } from '../../utils/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
  projects: createMockProjects,
  heading: 'List of Projects',
});

