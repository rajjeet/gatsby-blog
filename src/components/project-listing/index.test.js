import React from 'react';
import renderer from 'react-test-renderer';
import ProjectListing from './index';
import { makeProps } from './mock';

describe('<ProjectListing />', () => {
  it('should render', () => {
    const tree = renderer.create(<ProjectListing {...makeProps()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
