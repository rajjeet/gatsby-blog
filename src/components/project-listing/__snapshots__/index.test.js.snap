// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectListing /> should render 1`] = `
<div
  className="[object Object] sc-bxivhb nYNVZ"
>
  <h1>
    
  </h1>
  <div
    className="sc-htpNat cDaIEp"
  >
    <a
      aria-label="See Condo Landing Pages"
      className="sc-bwzfXH kTyWdp"
      href="/projects/condo-landing-pages/"
    >
      <h3>
        Condo Landing Pages
      </h3>
      <div
        className="sc-bdVaJa cBDXGb gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        
        <picture>
          <source
            sizes=""
            srcSet=""
          />
          <img
            alt="Condo Landing Pages"
            onLoad={[Function]}
            src=""
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"Condo Landing Pages\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <p>
        An automated process for building fast, responsive landing pages for pre-construction condos.
      </p>
    </a>
  </div>
  <div
    className="sc-htpNat cDaIEp"
  >
    <a
      aria-label="See Kitchen Quoter"
      className="sc-bwzfXH kTyWdp"
      href="/projects/kitchen-quoter/"
    >
      <h3>
        Kitchen Quoter
      </h3>
      <div
        className="sc-bdVaJa cBDXGb gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        
        <picture>
          <source
            sizes=""
            srcSet=""
          />
          <img
            alt="Kitchen Quoter"
            onLoad={[Function]}
            src=""
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"Kitchen Quoter\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <p>
        A do-it-yourself calculator that generates dollar quotes based on user-specified fields and formula.
      </p>
    </a>
  </div>
</div>
`;
